internal/resources/private_endpoint_service.go

```diff
@@ -0,0 +1,348 @@
+)
+
+type PrivateEndpointService struct {
+	*providerschema.Data
+}
+
+func NewPrivateEndpointService() resource.Resource {
+	return &PrivateEndpointService{}
+}
+
+func (p *PrivateEndpointService) Metadata(_ context.Context, req resource.MetadataRequest, resp *resource.MetadataResponse) {
+	resp.TypeName = req.ProviderTypeName + "_private_endpoint_service"
+}
+
+func (p *PrivateEndpointService) Schema(_ context.Context, _ resource.SchemaRequest, resp *resource.SchemaResponse) {
+	resp.Schema = schema.Schema{
+		Attributes: map[string]schema.Attribute{
+			"organization_id": stringAttribute([]string{required}),
+			"project_id":      stringAttribute([]string{required}),
+			"cluster_id":      stringAttribute([]string{required}),
+			"enabled":         boolAttribute(computed),
+		},
+	}
+}
+
+// Create enables private endpoint service.
+func (p *PrivateEndpointService) Create(ctx context.Context, req resource.CreateRequest, resp *resource.CreateResponse) {
+	var plan providerschema.PrivateEndpointService
+	diags := req.Plan.Get(ctx, &plan)
+	resp.Diagnostics.Append(diags...)
+	if resp.Diagnostics.HasError() {
+		return
+	}
+
+	err := validateCreateEndpointService(plan)
+	if err != nil {
+		resp.Diagnostics.AddError(
+			"Error validating private endpoint service request",
+			"Could not validate private endpoint service request, unexpected error: "+err.Error(),
+		)
+		return
+	}
+
+	var (
+		organizationId = plan.OrganizationId.ValueString()
+		projectId      = plan.ProjectId.ValueString()
+		clusterId      = plan.ClusterId.ValueString()
+	)
+
+	url := fmt.Sprintf(
+		"%s/v4/organizations/%s/projects/%s/clusters/%s/privateEndpointService",
+		p.HostURL,
+		organizationId,
+		projectId,
+		clusterId,
+	)
+
+	cfg := api.EndpointCfg{Url: url, Method: http.MethodPost, SuccessStatus: http.StatusAccepted}
+	_, err = p.Client.ExecuteWithRetry(
+		ctx,
+		cfg,
+		nil,
+		p.Token,
+		nil,
+	)
+	if err != nil {
+		resp.Diagnostics.AddError(
+			"Error enabling private endpoint service",
+			errorMessageWhileEnablingPrivateEndpointService+api.ParseError(err),
+		)
+		return
+	}
+
+	diags = resp.State.Set(ctx, initializePrivateEndpointServicePlan(plan))
+	resp.Diagnostics.Append(diags...)
+	if resp.Diagnostics.HasError() {
+		return
+	}
+
+	err = p.waitUntilStatusChanges(ctx, true, organizationId, projectId, clusterId)
+	if err != nil {
+		resp.Diagnostics.AddError(
+			"Error could not enable private endpoint service",
+			"Error could not enable private endpoint service, unexpected error: "+err.Error(),
+		)
+	}
+
+	refreshedState, err := p.getServiceState(ctx, organizationId, projectId, clusterId)
+	if err != nil {
+		resp.Diagnostics.AddError(
+			"Error reading private endpoint service status",
+			"Error reading private endpoint service status, unexpected error: "+err.Error(),
+		)
+
+		return
+	}
+
+	diags = resp.State.Set(ctx, refreshedState)
+	resp.Diagnostics.Append(diags...)
+	if resp.Diagnostics.HasError() {
+		return
+	}
+}
+
```

Comment:
Add GoDoc

Reply:
for the return statement ?  why ?

Reply:
Sorry for the confusion, I meant the GoDoc for the Read function.